xbrainmap/xbrain-py/code/celldetect/OMP_ProbMap.py

```diff
@@ -0,0 +1,165 @@
+# conditions are met:                                                     #
+#                                                                         #
+#     * Redistributions of source code must retain the above copyright    #
+#       notice, this list of conditions and the following disclaimer.     #
+#                                                                         #
+#     * Redistributions in binary form must reproduce the above copyright #
+#       notice, this list of conditions and the following disclaimer in   #
+#       the documentation and/or other materials provided with the        #
+#       distribution.                                                     #
+#                                                                         #
+#     * Neither the name of UChicago Argonne, LLC, Argonne National       #
+#       Laboratory, ANL, the U.S. Government, nor the names of its        #
+#       contributors may be used to endorse or promote products derived   #
+#       from this software without specific prior written permission.     #
+#                                                                         #
+# THIS SOFTWARE IS PROVIDED BY UChicago Argonne, LLC AND CONTRIBUTORS     #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT       #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS       #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL UChicago     #
+# Argonne, LLC OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,        #
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,    #
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;        #
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER        #
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT      #
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN       #
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE         #
+# POSSIBILITY OF SUCH DAMAGE.                                             #
+# #########################################################################
+
+"""
+Module for detecting cells from probablity map and return their centroids.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+
+# following imports to be updated when directory structure are finalized 
+from create_synth_dict import create_synth_dict
+from placeatom import roundno
+from convn_fft import convn_fft
+from compute3dvec import compute3dvec
+
+import numpy as np
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+__author__ = "Eva Dyer"
+__credits__ = "Mehdi Tondravi"
+
+__copyright__ = "Copyright (c) 2016, UChicago Argonne, LLC."
+__docformat__ = 'restructuredtext en'
+__all__ = ['OMP_ProbMap']
+
+
+
+def OMP_ProbMap(Prob, ptr, presid, startsz, dilatesz, kmax):
```

Comment:
- We need to rename functions with PEP8 compatible names. What does OMP mean? 
- Args should be lowercase, and something intuitive. Some suggestions: Prob -> cell_probability, ptr -> probability_threshold, presid -> stopping_criterion, startsz -> initial_template_size, dilatesz -> dilation_size, kmax -> max_no_cells